tests/pipeline/test_blaze.py

```diff
@@ -0,0 +1,455 @@
+from unittest import TestCase
+import warnings
+
+import blaze as bz
+from datashape import dshape, var, Record
+import numpy as np
+import pandas as pd
+from pandas.util.testing import assert_frame_equal
+from toolz import keymap
+from toolz.curried import operator as op
+
+from zipline.pipeline import Pipeline, CustomFactor
+from zipline.pipeline.data import DataSet, BoundColumn
+from zipline.pipeline.engine import SimplePipelineEngine
+from zipline.pipeline.loaders.blaze import (
+    pipeline_api_from_blaze,
+    BlazeLoader,
+    NoDeltasWarning,
+    NonNumpyField,
+    NonPipelineField,
+)
+from zipline.utils.test_utils import tmp_asset_finder
+
+
+nameof = op.attrgetter('name')
+dtypeof = op.attrgetter('dtype')
+
+
+class BlazeToPipelineTestCase(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.dates = pd.date_range('2014-01-01', '2014-01-03')
+        dates = (
+            [pd.Timestamp('2014-01-01')] * 3 +
+            [pd.Timestamp('2014-01-02')] * 3 +
+            [pd.Timestamp('2014-01-03')] * 3
+        )
+        cls.sids = sids = ord('A'), ord('B'), ord('C')
+        cls.df = df = pd.DataFrame({
+            'sid': sids * 3,
+            'value': tuple(
+                chain.from_iterable((a, a + 1, a + 2) for a in range(3)),
+            ),
+            'asof_date': dates,
+            'timestamp': dates,
+        })
+        cls.dshape = dshape_ = dshape("""
+        var * {
+            sid: ?int64,
+            value: ?float64,
+            asof_date: datetime,
+            timestamp: datetime
+        }
+        """)
+        cls.macro_df = df[df.sid == 65].drop('sid', axis=1)
+        dshape_ = OrderedDict(dshape_.measure.fields)
+        del dshape_['sid']
+        cls.macro_dshape = var * Record(dshape_)
+
+        cls.garbage_loader = BlazeLoader()
+
+    def test_tabular(self):
+        name = 'expr'
+        expr = bz.Data(self.df, name=name, dshape=self.dshape)
+        ds = pipeline_api_from_blaze(
+            expr,
+            loader=self.garbage_loader,
+            no_deltas_rule='ignore',
+        )
+        self.assertEqual(ds.__name__, name)
+        self.assertTrue(issubclass(ds, DataSet))
+        self.assertEqual(
+            {c.name: c.dtype for c in ds._columns},
+            {'sid': np.int64, 'value': np.float64},
+        )
+
+        for field in ('timestamp', 'asof_date'):
+            with self.assertRaises(AttributeError) as e:
+                getattr(ds, field)
+            self.assertIn("'%s'" % field, str(e.exception))
+            self.assertIn("'datetime'", str(e.exception))
+
+        self.assertIs(
+            pipeline_api_from_blaze(
+                expr,
+                loader=self.garbage_loader,
+                no_deltas_rule='ignore',
+            ),
+            ds,
+        )
+
+    def test_column(self):
+        exprname = 'expr'
+        expr = bz.Data(self.df, name=exprname, dshape=self.dshape)
+        value = pipeline_api_from_blaze(
```

Comment:
I think we can just call this function `from_blaze`.  I think it's clear from the import path that it's producing Pipeline objects.